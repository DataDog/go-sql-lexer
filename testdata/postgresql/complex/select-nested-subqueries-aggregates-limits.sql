SELECT 
  user_id,
  order_id,
  order_total,
  user_total
FROM (
  SELECT 
    o.user_id,
    o.id AS order_id,
    o.total AS order_total,
    (SELECT SUM(total) FROM orders WHERE user_id = o.user_id) AS user_total,
    RANK() OVER (PARTITION BY o.user_id ORDER BY o.total DESC) AS rnk
  FROM 
    orders o
) sub
WHERE 
  sub.rnk = 1
AND user_total > (
  SELECT 
    AVG(total) * 2 
  FROM orders
);
