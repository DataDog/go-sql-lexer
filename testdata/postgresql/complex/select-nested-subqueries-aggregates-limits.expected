[
    {
        "expected": "SELECT user_id, order_id, order_total, user_total FROM ( SELECT o.user_id, o.id, o.total, ( SELECT SUM ( total ) FROM orders WHERE user_id = o.user_id ), RANK ( ) OVER ( PARTITION BY o.user_id ORDER BY o.total DESC ) FROM orders o ) sub WHERE sub.rnk = ? AND user_total > ( SELECT AVG ( total ) * ? FROM orders )",
        "statement_metadata": {
            "size": 12,
            "tables": [
                "orders"
            ],
            "commands": [
                "SELECT"
            ],
            "comments": [],
            "procedures": []
        }
    }
]